src/App.jsx

```diff
@@ -6,14 +6,36 @@ function App() {
   const [monthlyContribution, setMonthlyContribution] = useState('')
   const [annualReturn, setAnnualReturn] = useState('7')
   const [yearsToRetirement, setYearsToRetirement] = useState('')
+  
+  // Real rate of return calculation inputs
+  const [nominalReturn, setNominalReturn] = useState('')
+  const [inflationRate, setInflationRate] = useState('')
+  
+  // Social security inputs
+  const [socialSecurityMonthly, setSocialSecurityMonthly] = useState('')
+  const [yearsUntilSocialSecurity, setYearsUntilSocialSecurity] = useState('')
+  
   const [results, setResults] = useState(null)
 
+  const calculateRealReturn = () => {
+    const nominal = parseFloat(nominalReturn) / 100 || 0
+    const inflation = parseFloat(inflationRate) / 100 || 0
+    
+    // Real rate = (1 + nominal) / (1 + inflation) - 1
+    const realRate = ((1 + nominal) / (1 + inflation) - 1) * 100
+    return realRate
+  }
+
   const calculateRetirement = () => {
     // Convert inputs to numbers
     const currentSavingsNum = parseFloat(currentSavings) || 0
     const monthlyContributionNum = parseFloat(monthlyContribution) || 0
     const annualReturnNum = parseFloat(annualReturn) / 100
     const yearsToRetirementNum = parseInt(yearsToRetirement) || 0
+    
+    // Social security inputs
```

Comment:
🧙‍♂️ **Senior Developer**

Why are we adding so many state variables in `App.jsx`? This reminds me of the last time someone thought every input needed its own useState. If this grows, it's going to be a nightmare to maintain. Consider grouping related state in a single object or using a reducer. Less is more. Also, there's no need for fancy hooks here. Simple is good. Please don't add Redux for a two-page app.